### PR TITLE
Adding cross domain signaling.

### PR DESCRIPTION
**Pull request by mindaugasbarcauskas**
Example use:

WorkflowExecution workflowExecution = new WorkflowExecution().setWorkflowId(workflowId);
      ExternalWorkflowStub workflowStub =
          Workflow.newUntypedExternalWorkflowStub(workflowExecution);
      workflowStub.signalCrossDomain("FareWellWorkflow::waitForName", "testdomain2", "Mindaugas");

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/sync/ExternalWorkflowStubImpl.java [hunk 1]
```diff
@@ -59,6 +59,24 @@ public void signal(String signalName, Object... args) {
     }
   }
 
+  @Override
+  public void signalCrossDomain(String signalName, String domain, Object... args) {
```
Comment (mindaugasbarcauskas) [inline src/main/java/com/uber/cadence/internal/sync/ExternalWorkflowStubImpl.java:63]: Probably similar change is needed in ChildWorkflowStub?

Comment (mindaugasbarcauskas) [reply src/main/java/com/uber/cadence/internal/sync/ExternalWorkflowStubImpl.java:63]: @Groxx maybe you know?

Comment (davidporter-id-au) [reply src/main/java/com/uber/cadence/internal/sync/ExternalWorkflowStubImpl.java:63]: I believe that's the case

**Review by davidporter-id-au — APPROVED** [review]
LGTM inasmuch as I'm pretty poor at reading Java, thank you

**Pull request closed without merging**